test/gtest/ucp/test_ucp_rma.cc

```diff
@@ -19,290 +23,143 @@ class test_ucp_rma : public test_ucp_memheap {
-        generate_test_params_variant(ctx_params, name, test_case_name, tls, 0,
-                                     result);
-        generate_test_params_variant(ctx_params, name, test_case_name + "/map_nb",
-                                     tls, UCP_MEM_MAP_NONBLOCK, result);
+        generate_test_params_variant(ctx_params, name, test_case_name + "/flush_ep",
+                                     tls, FLUSH_EP, result);
+        generate_test_params_variant(ctx_params, name, test_case_name + "/flush_worker",
+                                     tls, FLUSH_WORKER, result);
         return result;
     }
 
-    void nonblocking_put_nbi(entity *e, size_t max_size,
-                             void *memheap_addr,
-                             ucp_rkey_h rkey,
-                             std::string& expected_data)
-    {
-        ucs_status_t status;
-        status = ucp_put_nbi(e->ep(), &expected_data[0], expected_data.length(),
-                             (uintptr_t)memheap_addr, rkey);
-        ASSERT_UCS_OK_OR_INPROGRESS(status);
+    void put_b(size_t size, void *target_ptr, ucp_rkey_h rkey,
+               void *expected_data, void *arg) {
+        ucs_status_ptr_t status_ptr = do_put(size, target_ptr, rkey,
+                                             expected_data, arg);
+        request_wait(status_ptr);
     }
 
-    void nonblocking_put_nb(entity *e, size_t max_size,
-                            void *memheap_addr,
-                            ucp_rkey_h rkey,
-                            std::string& expected_data)
-    {
-        void *status;
-
-        status = ucp_put_nb(e->ep(), &expected_data[0], expected_data.length(),
-                            (uintptr_t)memheap_addr, rkey, send_completion);
-        ASSERT_UCS_PTR_OK(status);
-        if (UCS_PTR_IS_PTR(status)) {
-            request_wait(status);
-        }
+    void put_nbi(size_t size, void *target_ptr, ucp_rkey_h rkey,
+                 void *expected_data, void *arg) {
+        ucs_status_ptr_t status_ptr = do_put(size, target_ptr, rkey,
+                                             expected_data, arg);
+        request_release(status_ptr);
     }
 
-    void nonblocking_get_nbi(entity *e, size_t max_size,
-                             void *memheap_addr,
-                             ucp_rkey_h rkey,
-                             std::string& expected_data)
-    {
-        ucs_status_t status;
+    void get_b(size_t size, void *target_ptr, ucp_rkey_h rkey,
+               void *expected_data, void *arg) {
+        ucs_status_ptr_t status_ptr;
+        ucp_request_param_t param;
 
-        ucs::fill_random(memheap_addr, ucs_min(max_size, 16384U));
-        status = ucp_get_nbi(e->ep(), (void *)&expected_data[0], expected_data.length(),
-                             (uintptr_t)memheap_addr, rkey);
-        ASSERT_UCS_OK_OR_INPROGRESS(status);
+        param.op_attr_mask = 0;
+        status_ptr = ucp_get_nbx(sender().ep(), expected_data, size,
```

Comment:
there is common code between `get_b()` and `get_nbi()`, can be separate function `do_get()` as for PUT operation